.github/workflows/pr-merged.yml

```diff
@@ -0,0 +1,166 @@
+name: PR Merged
+
+on:
+  pull_request_target:
+    types: [closed]
+
+permissions:
+  contents: read
+
+jobs:
+  merged:
+    name: Show info
+    if: github.event.pull_request.merged == true
+    runs-on: ubuntu-24.04-arm
+    steps:
+      # github.event.pull_request.base may not refer to the _actual_ parent commit of the PR,
+      # instead it is the commit that the PR was merged/rebased onto.
+      #
+      # Since we want to diff the PR's commits, query the first commit's first parent.
+      #
+      # Alternatively, we could compare
+      # github.event.pull_request.merge_commit_sha to github.event.pull_request.base.sha,
+      # however I'm unsure how that'll interact with grouped Merge Queue 🤔
+      - name: Get base commit
+        id: base
+        env:
+          GH_TOKEN: ${{ github.token }}
+          owner: ${{ github.repository_owner }}
+          repo: ${{ github.event.repository.name }}
+          pr: ${{ github.event.pull_request.number }}
+        run: |
+          out=$(mktemp)
+          gh api graphql -F owner="$owner" -F repo="$repo" -F pr="$pr" -f query='
+              query($owner: String!, $repo: String!, $pr: Int!) {
+                repository(owner: $owner, name: $repo) {
+                  pullRequest(number: $pr) {
+                    commits(first: 1) {
+                      edges {
+                        node {
+                          commit {
+                            parents(first: 1) {
+                              edges {
+                                node {
+                                  oid
+                                }
+                              }
+                            }
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+              ' --jq '
+                .data
+                .repository
+                .pullRequest
+                .commits
+                .edges[0]
+                .node
+                .commit
+                .parents
+                .edges[0]
+                .node
+                .oid
+              ' > "$out"
+          echo "sha=$(cat "$out")" >> "$GITHUB_OUTPUT"
+
+      # Checkout CI scripts from the latest commit from the base branch.
+      # This ensures we get the latest scripts, even if the PR's base was outdated.
+      - name: Checkout base branch
+        id: checkout
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: flake/dev/diff-plugins.py
+          sparse-checkout-cone-mode: false
+
+      - name: Install Nix
+        uses: cachix/install-nix-action@v31
+        with:
+          nix_path: nixpkgs=channel:nixpkgs-unstable
```

Comment:
This is something I recently was migrating away from in Home Manager because it leads to an unstable CI. So maybe we'll want to tackle the `flake.lock` wrapper / shim sooner than later.

Reply:
Good point. Shouldn't be too complicated, depending on how closely we want to mirror how `call-flake.nix` evaluates a flake.lock. i.e. how many edge cases we want to handle (follows, `_2` inputs, etc).

I believe using flake-compat would lead to all inputs being downloaded, so we probably wouldn't want to do that in this case. But we could do something similar to how we _import_ flake-compat, i.e. read directly from the lockfile JSON.